Providers/PythonProvider.cpp

```diff
@@ -265,9 +265,21 @@ PythonProvider::~PythonProvider ()
     }
 }
 
+/* PythonProvider::init() creates a child process (referenced by m_pid) to execute dsc python resources using client.py script.
+   When execution of python resource (child process) is completed, it doesn’t disappear completely.
+   Instead it becomes Zombie which is not capable to execute anything.
+   Any further call to client.py (via socket) will fail.
+
+   The Zombie processes can be simply removed by registering for SIGCHLD signal and calling waitpid API from handler.
+
+   This method is a SIGCHLD handler, which is raised when a child process is terminated.
+*/
 void PythonProvider::handleChildSignal(int sig) {
     int saved_errno = errno;
-    // Child process is complete, release the process.
+
+    // Obtain information about the child whose state has changed, and release it.
+    // Only one instance of SIGCHLD can be queued, so it becomes necessary to reap
+    // several zombie processes during one invocation of the handler function.
```

Comment:
Since now we do waitpid here, do we still need the call in the destructor?

`
waitpid(m_pid, NULL, 0)`

Reply:
Yes, I am keeping that for the scenario where provider unload is called and child process has not finished the job. in that case provider should wait for the child process to finish.

---
In reply to: [143619410](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/331#discussion_r143619410) [](ancestors = 143619410)